### ISSUE TITLE
seat: keep track of what buttons are pressed

### ISSUE THREAD

**Issue by acrisci** [swaywm/wlroots#1077]
The seat pointer interface seems to be very prickly about how every button press must be followed by a button release.

For instance, if I just send a random `wlr_seat_notify_pointer_button()` with release, the `pointer_state.button_count` will now be wrong for the rest of the time that the compositor runs (with no buttons pressed, the value will be -1) because it simply decrements the button count whenever it gets a release.

I'm running into this now on #1018. Here, I don't want to send a button press when the compositor move starts. When the compositor move ends, it would be harmless to send a release for a button that was never pressed. However how it is now, that will completely screw up the seat state in a way that's difficult to debug and requires a compositor restart to fix.

The case of press without a release I would consider a developer error and we can't help that one.

**Comment by emersion** [comment]
Yeah, it's true that the pointer state is too error-prone. We should probably make it like the keyboard state.

**Comment by acrisci** [comment]
Another weird case we have to handle is if you press a button on the mouse, and then you disconnect your mouse, you have to release all the buttons that are pressed on the mouse.
